nifi-mock/src/main/java/org/apache/nifi/util/MockProcessorLog.java

```diff
@@ -16,20 +16,57 @@
  */
 package org.apache.nifi.util;
 
+import java.util.List;
+
 import org.apache.nifi.logging.ProcessorLog;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class MockProcessorLog implements ProcessorLog {
 
-    private final Logger logger;
+    private final CapturingLogger logger;
     private final Object component;
 
     public MockProcessorLog(final String componentId, final Object component) {
-        this.logger = LoggerFactory.getLogger(component.getClass());
+        this.logger = new CapturingLogger(LoggerFactory.getLogger(component.getClass()));
```

Comment:
Can you explain why we have this CapturingLogger here, which implements many methods, rather than just modifying the existing warn(), error(), etc. methods? It seems that the capturing could happen within the MockProcessorLog, rather than at a lower level. I feel like this would result in much simpler code. But it's quite possible that I am missing something? Let me know what you think!